sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -206,17 +203,19 @@ public virtual Response Upload<T>(string ruleId, string streamName, IEnumerable<
             Argument.AssertNotNullOrEmpty(logs, nameof(logs));
 
             using var scope = ClientDiagnostics.CreateScope("LogsIngestionClient.Upload");
-
             Response response = null;
             List<Exception> exceptions = null;
-
             scope.Start();
+
             // Keep track of the number of failed logs across batches
             int logsFailed = 0;
-
+            var shouldAbort = false;
             // Partition the stream into individual blocks
-            foreach (BatchedLogs<T> batch in Batch(logs, options))
+            foreach (BatchedLogs batch in Batch((List<object>)logs, options))
```

Comment:
Why are we casting to a list?

Reply:
The helper method Batch no longer has type <T>, let me figure out why I removed it